### DISCUSSION TITLE
POL 2 Island of Chaos - mission is impossible (red and yellow are allied to each other)

### DISCUSSION THREAD

**Discussion by hpta** [ihhub/fheroes2#9010] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [ ] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

AI just snowballs and cannot be stopped. Apparently in the original game red and yellow were enemies.

It's mentioned in the comments here https://www.youtube.com/watch?v=bSq-PlmPzy4

"hi i just tried Fheroes, nope i couldn;t beat it. I noticed red and yellow are avoiding each other instead of beating each other. That might be why this is difficult to beat"
"Wait, I thought red and yellow are allies (I tried a lot, and I never saw them attacking each other). In the original version, they are enemies? If so, this sounds like a bug of fheroes."

### Save file

[POL_2_0016.zip](https://github.com/user-attachments/files/16403794/POL_2_0016.zip)

**Comment by zenseii** [comment]
I'm closing this and converting this to a discussion because the map can be defeated on latest patch on the hardest difficulty as proven by other players in the community.

@dood-apo made a post about how to beat it on our Discord server just two days ago.

**Reply by CommonLoon102** [reply]
@dood-apo @zenseii could you please post it here also? With fheroes2, this map seems to be impossible, even on normal difficulty. Others are complaining too.
Is it still beatable on 1.1.16?

**Reply by CommonLoon102** [reply]
It is impossible on fheroes now: https://www.youtube.com/watch?v=D7_y3ISRqIE&list=PL3cZlWKHw-UDr3LPn4iuiWzhpeNBjB4HM&index=2

**Reply by CommonLoon102** [reply]
There is a video for 1.1.8 where the map is completed: https://www.youtube.com/watch?v=_Za8Ffgp4Ho
But no video for 1.1.16. Is there some kind of vanilla mode where the AI enhancements can be turned off?

**Reply by zenseii** [reply]
@CommonLoon102. No. The map is still beatable. Here is a video of it being beat on Hard from last August: https://youtu.be/ZUi9EtjR1Vg?si=eTl76GnRaBg_NKY9
None of the changes to the AI since then will have had any impact on the map's completability. Certainly not making it go from beatable on hard to impossible on normal.

**Reply by zenseii** [reply]
@CommonLoon102 Here's a playthrough on Hard on the very latest 1.1.17
https://youtu.be/J8zGBsMuLO8?t=1995&is=uMadOHMgmLzTEZH7

**Comment by zenseii** [comment]
As for the comment about how the red and yellow players are supposed to be enemies, this is not true. This original campaign's map file contains an alliance between all the AI players.

The reason someone might misunderstand their alliance is because in the original game you could attack your own ally and flag their mines, and this is something the original AI had a tendency to do when it was pathing. We will not re-implement this behavior in fheroes2 since it is far from good.

**Reply by oleg-derevenetz** [reply]
In addition, in the original HoMM2, allies did not have a shared map vision, therefore, when one of the allies already knew about you, the other might not know yet. In fheroes2, allies have a shared map vision, therefore, AI-controlled allies can attack the enemy more aggressively. Although of course there is an opinion that in the original game AI eventually got the ability to see the entire map anyway.
